src/ResourceManagement/MachineLearning/Microsoft.Azure.Management.MachineLearning/Management/WebService.cs

```diff
@@ -0,0 +1,273 @@
+        {
+            this.Definition = definitionWebService;
+            this._client = client;
+            this.ResourceGroupName = resourceGroup;
+        }
+
+        /// <summary>
+        /// Reads in a web service from a web service definition json file.
+        /// </summary>
+        /// <param name="webServiceParameters">Parameters to replace within the file if interpolation parameters exist.</param>
+        /// <param name="webServiceDefinitionFilePath">The path to the web service definition file.</param>
+        /// <returns></returns>
+        private Microsoft.Azure.Management.MachineLearning.WebServices.Models.WebService InputWebServiceFromWSDFile(Dictionary<string, string> webServiceParameters, string webServiceDefinitionFilePath)
+        {
+            string wsDefinition;
+
+            //StreamReader r = new StreamReader("file.txt"); System.IO.
+            using (StreamReader r = System.IO.File.OpenText(webServiceDefinitionFilePath))
+            {
+                wsDefinition = r.ReadToEnd();
+            }
+
+            foreach (var kvp in webServiceParameters)
+            {
+                wsDefinition = wsDefinition.Replace($"$({kvp.Key})", kvp.Value);
+            }
+
+            // TODO: call this in the constructor and assign 
+            return ModelsSerializationUtil.GetAzureMLWebServiceFromJsonDefinition(wsDefinition);
+        }
+
+        /// <summary>
+        /// Updates this web service and updates the internal state.
+        /// </summary>
+        /// <param name="otherWebService">The other web service from which this web service is being updated.</param>
+        public void Update()
+        {
+            Microsoft.Azure.Management.MachineLearning.WebServices.Models.WebService ws = this._client.WebServices.Patch(this.Definition, this.ResourceGroupName, this.Title);
+
+            this.Definition = ws;
+        }
+
+        /// <summary>
+        /// Updates this web service and updates the internal state. Takes an AutoRest web service model.
+        /// </summary>
+        /// <param name="otherWebService">The other AutoRest web service model from which this web service is being updated.</param>
+        internal void Update(Microsoft.Azure.Management.MachineLearning.WebServices.Models.WebService otherWebService)
+        {
+            Microsoft.Azure.Management.MachineLearning.WebServices.Models.WebService ws = this._client.WebServices.CreateOrUpdate(otherWebService, this.ResourceGroupName, this.Title);
+
+            this.Definition = ws;
+        }
+
+        /// <summary>
+        /// Deletes this deployed web service.
+        /// </summary>
+        public void Delete()
+        {
+            this._client.WebServices.Remove(this.ResourceGroupName, this.Title);
+        }
+
+        /// <summary>
+        /// Synchronizes the internal state keys with the keys on the server side.
+        /// </summary>
+        public void SyncKeys()
```

Comment:
From the implementation, you sync "down" the key. So you probably want to make the name of method clear "SyncKeysFromCloud" or something like this.